<launch>

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find crustcrawler_control)/config/crustcrawler_control.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
	output="screen" ns="/crustcrawler" args="joint_state_controller
					  joint_1_position_controller
					  joint_2_position_controller
					  joint_3_position_controller
					  joint_4_position_controller
					  joint_5_position_controller
					  joint_6_position_controller
					  joint_right_finger_position_controller
					  joint_left_finger_position_controller"/>


  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
    <!--param name="robot_description" command="$(find xacro)/xacro.py '$(find crustcrawler_description)/urdf/crustcrawler.xacro'" /-->
    <remap from="/joint_states" to="/crustcrawler/joint_states" />
  </node>
  
  <node name="crustcrawler_translator" pkg="crustcrawler_control" type="crustcrawler_command_translator" respawn="false" output="screen"/>
  
  <node name="crustcrawler_gripper" pkg="crustcrawler_control" type="crustcrawler_gripper" respawn="false" output="screen"/>

</launch>
